backends/p4test/midend.cpp

```diff
@@ -141,6 +142,7 @@ MidEnd::MidEnd(CompilerOptions& options, std::ostream* outStream) {
         new P4::SynthesizeActions(&refMap, &typeMap, new SkipControls(v1controls)),
         new P4::MoveActionsToTables(&refMap, &typeMap),
         options.loopsUnrolling ? new P4::ParsersUnroll(true, &refMap, &typeMap) : nullptr,
+        options.hsIndexSimplify ? new P4::HSIndexSimplifier(&refMap, &typeMap) : nullptr,
```

Comment:
I personally think to properly test it, this pass should be used on the entire test suite. We do not have the same restrictions as with loops unrolling.